api/spec/v1.yaml

```diff
@@ -767,6 +769,71 @@ paths:
               schema:
                 $ref: '#/components/schemas/RuntimeTransactionList'
         <<: *common_error_responses
+  
+  /emerald/events:
+    get:
+      summary: Returns a list of Emerald events.
+      parameters:
+        - *limit
+        - *offset
+        - in: query
+          name: block
+          schema:
+            type: integer
+            format: int64
+          description: A filter on block round.
+          example: *runtime_block_round_1
+        - in: query
+          name: tx_index # TODO: Is it possible to get the tx_index from the oasis-sdk (currently only returns txhash)? Could make a second call to fetch the tx by its hash...
+          schema:
+            type: integer
+            format: int32
+          description: |
+            A filter on transaction index. The returned events all need to originate
+            from a transaction that appeared in `tx_index`-th position in the block.
+            It is invalid to specify this filter without also specifying a `block`.
+            Specifying `tx_index` and `round` is an alternative to specifying `tx_hash`;
+            either works to fetch events from a specific transaction.
+          example: 3
+        - in: query
+          name: tx_hash
+          schema:
+            type: string
+          description: |
+            A filter on the hash of the transaction that originated the events.
+            Specifying `tx_index` and `round` is an alternative to specifying `tx_hash`;
+            either works to fetch events from a specific transaction.
+          example: *tx_hash_1
+        - in: query
+          name: rel
+          schema:
+            type: string
+          description: |
+            A filter on related accounts. Every returned event will refer to
+            this account. For example, for a `accounts.Transfer` event, this will be
+            the sender or the recipient of tokens.
+          example: *staking_address_1
+        - in: query
+          name: type
+          schema:
+            $ref: '#/components/schemas/RuntimeEventType'
+          description: A filter on the event type.
+        - in: query
+          name: evm_log_name
```

Comment:
I expect people will more so want to filter by `evm_event_topic`, i.e. the hash of the event type signature. That's as close as we can get to "event type" in EVM. Also, the topic is always available; the name is not, because we have to first know the ABI to parse the event to derive the name. So even for unparsed events, it will be nice to have a link like "show me other events of this type" in the GUI. It's also more selective than just the name.

To index it, you can stuff it in a separate field in the DB:
```sql
indexer=# create table foo(j json);
CREATE TABLE
indexer=# insert into foo values ('{"topic": "bar"}'::json);
INSERT 0 1
indexer=# insert into foo values ('{"not_a_topic": "bar"}'::json);
INSERT 0 1
indexer=# select j->'topic' from foo;
 ?column? 
----------
 "bar"
 
(2 rows)



indexer=# create table foo2 (j json, x text generated always as (j->'topic') stored);
CREATE TABLE
indexer=# insert into foo2(j) values ('{"not_a_topic": "bar"}'::json);
INSERT 0 1
indexer=# insert into foo2(j) values ('{"topic": "barzz"}'::json);
INSERT 0 1
indexer=# select * from foo2;
           j            |    x    
------------------------+---------
 {"not_a_topic": "bar"} | 
 {"topic": "barzz"}     | "barzz"
(2 rows)

indexer=# create index adfasdggg on foo2 (x);
CREATE INDEX
```

Reply:
Got it, updated. 

Note: I chose not to extract+index the other possible event topics because they are a subset of the `evm_log_params` which is indexed